### ISSUE TITLE
Pricing Settings - Prevent currency modification when already used!

### ISSUE THREAD

**Issue by ClaudeROSSI** [sap-labs-france/ev-server#3241]
The pricing and the billing of transactions is very sensitive to the actual currency defined at the pricing settings level.

We need to make sure the currency cannot be changed by mistake. Changing the currency should be possible but with a careful handling of existing data.

As a first step, we could proceed as this is done for the Billing settings. We could have a button in the UI to explicitly validate and activate the pricing settings. Once this is done, the currency cannot be changed anymore and tariffs can then be created.

In order to achieve this, it makes sense to add endpoints dedicated to the pricing settings (as this is already done for the billing).

**Comment by ClaudeROSSI** [comment]
The pricing settings logic has been changed to make sure the currency code cannot be changed too easily.

The currency code can be set only once.
- Once it is set the input field is then in a read-only mode.
- This only applies when using our Built-in Pricing Engine".

The new logic is currently implemented client-side by comparing the actual currency code (from the user token) with the one defined in the pricing settings.
No logic to prevent a currency change is right now in place on the server-side.

**Comment by ClaudeROSSI** [comment]
Closing this one because the super admin still needs to be able to change the currency.
So the setting endpoint does not prevent a modification.

The server-side has been changed (in another PR)to force the user to log out when the currency is being set.
After saving the input field is disabled.
